spec/support/view_component/matchers.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Support
+  module ViewComponent
+    module Matchers
+      extend RSpec::Matchers::DSL
+
+      define :have_rendered_component do |component_class|
+        match do |rendered|
+          @stubs = _test_context.components.select { |stubbed| stubbed.instance.is_a? component_class }
+          @matching = @stubs.select { |stubbed| arguments.args_match?(*stubbed.args) }
+          @matching.any? do |stubbed|
+            rendered.content.include?(stubbed.output)
```

Comment:
Is that making use of the lorem ipsums above? Can it find a wrong stubbed component if their output is the same?

Reply:
Yes, context stores all the outputs and uses them in the matcher to find that the component has in fact been rendered. If we had two components that has the same output then it could, theoretically, find a wrong component in the output. However the chance of this happening is infinitesimal - I haven't got a single duplicate in 1,000,000 random paragraphs, and we're using `paragraphs` which by default uses 4 paragraphs. I'd say probability would be comparable with uuid colliding.

Reply:
Absolutely, uuid will do the trick (should be also slightly faster). :)